.circleci/config.yml

```diff
@@ -1,70 +1,129 @@
-version: 2
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: circleci/node:latest
 
 jobs:
-  build:
-    docker:
-      - image: circleci/node:latest
+  init:
+    <<: *defaults
     steps:
       - checkout
       - run:
-          name: Update npm
-          command: 'sudo npm install -g npm@latest'
-      - restore_cache:
-          key: dependency-cache-{{ checksum "package.json" }}
-      - run:
-          name: NPM Install
-          command: npm install
+          name: npm install
+          command: npm i
       - run:
-          name: NPM Build
-          command: npm run build
+          name: install ci tools
+          command: 'mkdir -p ~/.build_tools/node && npm install --prefix ~/.build_tools/node kl-ci-tools'
+      - run: |
+          if [[ "$CIRCLE_BRANCH" =~ release/v.* ]]; then
+            ~/.build_tools/node/node_modules/kl-ci-tools/dist/bin.js deps | xargs -I[] npm install []@latest
+          fi
       - save_cache:
-          key: dependency-cache-{{ checksum "package.json" }}
+          key: git-sha-{{ .Revision }}{{ .Branch }}
           paths:
+            - package.json
             - node_modules
-            - dist
+            - ~/.build_tools
   test:
-    docker:
-      - image: circleci/node:latest
+    <<: *defaults
     steps:
       - checkout
       - restore_cache:
-          key: dependency-cache-{{ checksum "package.json" }}
+          key: git-sha-{{ .Revision }}{{ .Branch }}
       - run:
-          name: Lint
+          name: npm build
+          command: npm run build
+      - run:
+          name: lint
           command: npm run lint
       - run:
-          name: Test
+          name: test
           command: npm test
+      - save_cache:
```

Comment:
It looks like this step is not working since circleci does not allow overwriting the cache key.  

Example CircleCI output from this build here:
https://app.circleci.com/pipelines/github/kochavalabs/xdr-js-serialize/21/workflows/ba0e179c-0955-4fc7-b0be-f4c98fe27294/jobs/54/parallel-runs/0/steps/0-106

Should the npm build step be added to init maybe so the dist path can be included in the first cache?

Reply:
I just consolidated to one cache step.